src/main/java/uk/gov/companieshouse/confirmationstatementapi/service/FilingService.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.companieshouse.confirmationstatementapi.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.filinggenerator.FilingApi;
+import uk.gov.companieshouse.confirmationstatementapi.exception.SubmissionNotFoundException;
+import uk.gov.companieshouse.confirmationstatementapi.model.json.ConfirmationStatementSubmissionJson;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class FilingService {
+
+    @Value("${CONFIRMATION_STATEMENT_NO_UPDATES}")
+    private String filingDescription;
+
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd");
+    private ConfirmationStatementService confirmationStatementService;
+
+    @Autowired
+    public FilingService(ConfirmationStatementService csService) {
+        confirmationStatementService = csService;
+    }
+
+    public FilingApi generateConfirmationFiling(String confirmationStatementId) throws SubmissionNotFoundException {
+        FilingApi filing = new FilingApi();
+        filing.setKind("Confirmation Statement");
+        setFilingApiData(filing, confirmationStatementId);
+        setDescription(filing);
+        return filing;
+    }
+
+    private void setFilingApiData(FilingApi filing, String confirmationStatementId) throws SubmissionNotFoundException {
+        Optional<ConfirmationStatementSubmissionJson> submissionOpt =
+                confirmationStatementService.getConfirmationStatement(confirmationStatementId);
+        ConfirmationStatementSubmissionJson submission = submissionOpt
+                .orElseThrow(() ->
+                        new SubmissionNotFoundException(
+                                String.format("Empty submission returned when generating filing for %s", confirmationStatementId)));
+        if (submission.getData() != null) {
+            Map<String, Object> data = new HashMap<>();
+            data.put("confirmationStatementDate", submission.getData().getMadeUpToDate());
```

Comment:
it is probably worth making the maps keys snake_case to match the field names

Reply:
I just put them in camel case to match the xml examples

Reply:
The converter will have to convert from json to xml anyway. But it doesn't really make much difference, you can leave it as is. If it becomes an issue we can always come back and change